py3status/modules/wifi.py

```diff
@@ -136,11 +132,20 @@ def _dbm_to_percent(self, dbm):
     def _percent_to_dbm(self, percent):
         return (percent / 2.0) - 100
 
+    def _get_wifi_data(self, command):
+        for time in range(2):
+            try:
+                return self.py3.command_output(command)
+            except self.py3.CommandError as ce:
```

Comment:
I'm worried that CommandError captures more than just "operation not permitted" errors, so we will end up re-trying with `sudo` commands that failed for completely different reason, and then we will mislead users with a message about "sudo: password missing" instead of telling them that command failed

Reply:
My understanding is it should do this...
* Fail on first try, add `sudo`, and fail again, show `sudo` message.
* Run fine on first try (with or without `sudo`) and fail later, show that error.
* Because of `2`, it should show us that error... only that we wanted users to try it again with `sudo` first to confirm if it's still throwing an error unrelated to `sudo`.

Reply:
My point is, let's say there's some kind of error on the first try (maybe even uninstalled "iw" will work as a test case), in this PR we will try to run `iw`, then `sudo iw`, then fail and say "sudo needs your password". But it's misleading because the actual mistake is completely elsewhere.

Reply:
We check for `iw` first so we already know we can use `iw` or we throw exception.

Without that exception, `iw` gives us `None` and it fails inside `self.py3.command_output`. Any other commands that ran fine (first time) will be added to `storage` so if an error throw later, we know not to add `sudo` there since it's already in `storage`.

If we use random commands like `grrrr`, it would be tested then again with `sudo`, then we show `plz try again with sudo` first (maybe misleading but is to rule `sudo` issue out)... which could happen to commands `iw` and `ip` so we're checking only those two and not `grrrr`.

For `grrrr` and other `no-sudo` commands, we stay with normal `self.py3.command_output`.

This can be misleading if users don't need `sudo`, but ran into issues with their `iw` for some unknown / rare reasons... which we would try again with `sudo` then password.

I don't know how to differentiate this and I hadn't used `iw` without `use_sudo` before. Do you require `use_sudo` for this module on Arch Linux?

Reply:
Have a play with this. See if you can come up with something different. I think `iw` and `ip` wouldn't throw errors much and this is done to rule out `sudo`-related issues first for `iw` and `ip` before opening an issue here. Otherwise, maybe users have a serious issue somewhere if errors are still being thrown.

Reply:
Arch Linux doesn't require `use_sudo` — OK, I agree that the chances of having users struggle with not working module because they are missing `use_sudo` is much higher than some theoretical issues with `iw`, therefore I'm fine with merging this PR, we can always adapt again if our assumptions turn out to be incorrect.

Reply:
I added `no available device` exception to ensure that we're not going to have some `iw` theoretical issues because with `iw dev`, we can confirm this device exists first before continuing... and that it prints out okay first thing first.

Otherwise, I find that `iw dev` can throw an error -- which should address your `iw` theoretical issues unrelated to `sudo`. I assume anything after that, with confirmed device, would require nothing more than requiring users to use `sudo` because of a partial error with an output.

It is possible that `sudo` for `ip addr` was added for consistency purpose / to be utilized with `use_sudo` config too because I hadn't found a linux distribution that explicitly requires it. The commit history seems to indicate either that or Fedora do needs it. This is not ruled out anyway.